integration_test/src/test_runner/mod.rs

```diff
@@ -0,0 +1,327 @@
+/// names.
+pub fn load_file_paths(dir: PathBuf) -> Result<Vec<String>, Error> {
+    let mut paths: Vec<String> = vec![];
+    let entries = fs::read_dir(dir).map_err(|e| Error::InvalidDirectory(format!("{:?}", e)))?;
+    for entry in entries {
+        match entry {
+            Ok(de) => {
+                let path = de.path();
+                if (path.extension().unwrap() == "yml") || (path.extension().unwrap() == "yaml") {
+                    paths.push(path.to_str().unwrap().to_string());
+                }
+            }
+            Err(e) => return Err(Error::InvalidFilePath(format!("{:?}", e))),
+        };
+    }
+    Ok(paths)
+}
+
+/// parse_test_file_yaml deserializes the given YAML file into a
+/// IntegrationTest struct.
+pub fn parse_test_file_yaml(path: &str) -> Result<IntegrationTest, Error> {
+    let mut f = fs::File::open(path).map_err(|e| Error::InvalidFile(format!("{:?}", e)))?;
+    let mut contents = String::new();
+    f.read_to_string(&mut contents)
+        .map_err(|e| Error::CannotReadFileToString(format!("{:?}", e)))?;
+    let yaml: IntegrationTest = serde_yaml::from_str(&contents)
+        .map_err(|e| Error::CannotDeserializeYaml(format!("{:?}", e)))?;
+    Ok(yaml)
+}
+
+/// connect obtains a connection to the local Atlas Data Federation instance and returns Connection
+/// it fetches environment variables to create the connection string
+fn connect() -> Result<Connection<'static>, Error> {
+    let user_name = env::var("ADL_TEST_USER").expect("ADL_TEST_USER is not set");
+    let password = env::var("ADL_TEST_PWD").expect("ADL_TEST_PWD is not set");
+    let host = env::var("ADL_TEST_HOST").expect("ADL_TEST_HOST is not set");
+    let auth_db = match env::var("ADL_TEST_AUTH_DB") {
+        Ok(val) => val,
+        Err(_) => "admin".to_string(), //Default auth db
+    };
+    let db = match env::var("ADL_TEST_DB") {
+        Ok(val) => val,
+        Err(_) => "INTEGRATION_TEST".to_string(), //Default driver name
+    };
+    let driver = match env::var("ADL_TEST_DRIVER") {
+        Ok(val) => val,
+        Err(_) => "ADL_ODBC_DRIVER".to_string(), //Default driver name
+    };
+
+    let connection_string = format!(
+        "Driver={};PWD={};USER={};SERVER={};AUTH_SRC={};Database={}",
+        driver, password, user_name, host, auth_db, db
+    );
+    let connection = ODBC_ENV
+        .connect_with_connection_string(&connection_string)
+        .unwrap();
+
+    assert!(!connection.is_dead().unwrap());
+    Ok(connection)
+}
+
+/// integration_test runs the query and function tests contained in the TEST_FILE_DIR directory
+#[test]
+#[ignore]
+pub fn integration_test() -> Result<(), Error> {
+    let paths = load_file_paths(PathBuf::from(TEST_FILE_DIR)).unwrap();
+    for path in paths {
+        let yaml = parse_test_file_yaml(&path).unwrap();
+
+        for test in yaml.tests {
+            match test.skip_reason {
+                Some(_) => continue,
```

Comment:
Should we print/log some indication of the test being skipped, similar to how we do for the mongosql-rs spec tests?

Reply:
Sure, I added a print of the skip reason.